### PR TITLE
Add node global transform definition

### PR DESCRIPTION
**Pull request by lexaknyazev**
Fixes https://github.com/KhronosGroup/glTF/issues/1823.

### CODE REVIEW

**Review by emackey — APPROVED** [review]

**Comment by javagl** [comment]
Hm...

> The global transformation matrix of a node is the product of its local transformation matrix and the local transformation matrices of all its parent nodes.

A node has at most _one_ parent. One could rephrase it to "...all its ancestor nodes" to avoid this. But considering the importance of the _order_ of matrix multiplications, and that the statement, when interpreted verbatim, suggests a wrong order (it's `parent.transform * node.transform`, and not `node.transform * parent.transform`), one could consider phrasing that a bit differently. Either as in the overview,

> The global transform of a node is given by the product of all local transforms on the path from the root to the respective node.

or a bit more elaborately

> The global transformation matrix of a node is the product the global transformation matrix of its parent, and its local transformation matrix.  When the node has no parent node, then its global transformation matrix is equal to its local transformation matrix.

(or even "...identical", FWIW...)
